public/app/core/services/backend_srv.ts

```diff
@@ -53,6 +64,38 @@ export class BackendSrv implements BackendService {
   }
 
   fetch<T>(options: BackendSrvRequest): Observable<FetchResponse<T>> {
+    return new Observable(observer => {
+      // We need to match an entry added to the queue stream with the entry that is eventually added to the response stream
+      // using Date.now() as the unique identifier
+      const id = Date.now().toString(10);
+
+      // Subscription is an object that is returned whenever you subscribe to an Observable.
+      // You can also use it as a container of many subscriptions and when it is unsubscribed all subscriptions within are also unsubscribed.
+      const subscriptions: Subscription = new Subscription();
+
+      // We're using the subscriptions.add function to add the subscription implicitly returned by this.responseQueue.getResponses<T>(id).subscribe below.
+      subscriptions.add(
+        this.responseQueue.getResponses<T>(id).subscribe(result => {
+          // The one liner below can seem magical if you're not accustomed to RxJs.
+          // Firstly, we're subscribing to the result from the result.observable and we're passing in the outer observer object.
+          // By passing the outer observer object then any updates on result.observable are passed through to any subscriber of the fetch<T> function.
+          // Secondly, we're adding the subscription implicitly returned by result.observable.subscribe(observer).
+          subscriptions.add(result.observable.subscribe(observer));
```

Comment:
black magic! good that you added the comment here :) 
Can't say I understand this 100% yet :)

Reply:
Thank you for reviewing this, I tried to comment as much as possible as I know how hard it is to reason about RxJs when you're not an expert on RxJs :)